### PR TITLE
Remove flow from lib

### PR DESCRIPTION
**Pull request by ranquild**
Epic: https://github.com/metabase/metabase/issues/19207

### CODE REVIEW

File: frontend/src/metabase/lib/entities.js [hunk 1]
```diff
@@ -30,160 +30,10 @@ import _ from "underscore";
...
-
-  getObjectStatePath: Function,
-  getListStatePath: Function,
-
-  HACK_getObjectFromAction: (action: Action) => any,
-};
-
-export function createEntity(def: EntityDefinition): Entity {
-  const entity: Entity = { ...def };
+// FIXME
```
Comment (daltojohnso) [inline frontend/src/metabase/lib/entities.js:33]: let's remove this comment

File: frontend/src/metabase/lib/formatting/date.js [hunk 2]
```diff
@@ -1,26 +1,8 @@
 import { parseTimestamp } from "metabase/lib/time";
-import type { DateSeparator } from "metabase/lib/formatting";
 
-import type { DatetimeUnit } from "metabase-types/types/Query";
+// moment.js format strings
```
Comment (daltojohnso) [inline frontend/src/metabase/lib/formatting/date.js:3]: let's remove this comment

**Review by daltojohnso — APPROVED** [review]
look's good. a few random comments related to the typing code were left in. mind removing them?

File: frontend/src/metabase/lib/entities.js (hunk 1 above)
Comment (ranquild) [reply frontend/src/metabase/lib/entities.js:33]: Sure, I'll remove them in another open PR

**Pull request merged**